src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -45,24 +46,23 @@ public GitHubPackage(IServiceProvider serviceProvider)
             this.serviceProvider = serviceProvider;
         }
 
-        protected override void Initialize()
+        protected override async tasks.Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
-            IncrementLaunchCount();
+            var usageTracker = await GetServiceAsync(typeof(IUsageTracker)) as IUsageTracker;
+            await tasks.Task.Run(() => usageTracker.IncrementLaunchCount());
+            var menus = await GetServiceAsync(typeof(IMenuProvider)) as IMenuProvider;
+
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            var menus = serviceProvider.GetExportedValue<IMenuProvider>();
             foreach (var menu in menus.Menus)
                 serviceProvider.AddCommandHandler(menu.Guid, menu.CmdId, (s, e) => menu.Activate());
 
             foreach (var menu in menus.DynamicMenus)
                 serviceProvider.AddCommandHandler(menu.Guid, menu.CmdId, menu.CanShow, () => menu.Activate());
-        }
 
-        void IncrementLaunchCount()
-        {
-            var usageTracker = serviceProvider.GetExportedValue<IUsageTracker>();
-            usageTracker.IncrementLaunchCount();
+            await base.InitializeAsync(cancellationToken, progress);
```

Comment:
I think the recommended pattern is for you to call the base class method before your own implementation.